src/app/client/projects/desktop/src/app/modules/offline/components/library/library.component.ts

```diff
@@ -23,24 +29,144 @@ export class LibraryComponent implements OnInit {
     public dataDrivenFilterEvent = new EventEmitter();
     public unsubscribe$ = new Subject<void>();
 
-    public contentList = [];
-    public subjects = ['english', 'mathematics', 'geology', 'biology', 'zoology', 'Botany', 'Environmental Science'];
+    public noResultMessage: INoResultMessage;
+
+    slideConfig = this.configService.appConfig.CourseBatchPageSection.slideConfig;
 
 
+    @HostListener('window:scroll', []) onScroll(): void {
+        if ((window.innerHeight + window.scrollY) >= (document.body.offsetHeight * 2 / 3)
+            && this.pageSections.length < this.carouselMasterData.length) {
+            this.pageSections.push(this.carouselMasterData[this.pageSections.length]);
+        }
+    }
     constructor(
-        public resourceService: ResourceService,
+        private activatedRoute: ActivatedRoute,
+        private router: Router,
+        private pageApiService: PageApiService,
+        private utilService: UtilService,
+        private toasterService: ToasterService,
+        private configService: ConfigService,
+        private resourceService: ResourceService,
+        private orgDetailsService: OrgDetailsService
     ) { }
 
     ngOnInit() {
+        this.setNoResultMessage();
```

Comment:
resource service will get data by making api call. If the data has not fetched when this method is called all properties will be set as undefined or we might get error. And also this method is not considering language change. Pass this data to no result component in html it self